### ISSUE TITLE
Remove usage of `OpenStruct` or add `ostruct` as dependency 

### ISSUE THREAD

**Issue by Earlopain** [mileszs/wicked_pdf#1116]
## Issue description

Hey there, I saw https://github.com/mileszs/wicked_pdf/pull/1110 which added a require of `ostruct` to work around an error.

`ostruct` is on its way to be a bundled gem and Ruby 3.4 will warn if it is required without being specified as a dependency. https://github.com/ruby/ruby/pull/10428

So I suggest removing the usage to avoid breaking in Ruby 3.5. Adding it to the gemspec is also a solution.

**Comment by joshRpowell** [comment]
agree. `jbuilder` removed `ostruct` https://github.com/rails/jbuilder/pull/567/files

**Comment by Earlopain** [comment]
https://github.com/mileszs/wicked_pdf/commit/36cf39c3d3d4191d0a9334005a3cb28b11bb97ee
